src/actions/heal.cpp

```diff
@@ -212,7 +214,10 @@ namespace {
 
 		// Now we can get the aggregate healing amount.
 		unit_abilities::effect heal_effect(heal_list, 0, false);
-		if ( update_healing(healing, harming, heal_effect.get_composite_value()) )
+		if (heal_list.lowest("value").first < 0) healer_down = heal_list.lowest("value").first;
+		if (heal_effect.get_composite_value() >= 0) healer_up = heal_effect.get_composite_value();
+		int heal_value = healer_up + healer_down;
+		if ( update_healing(healing, harming, heal_value) )
```

Comment:
Why aren't the first two conditions symmetric? That is, why does the condition use get_composite_value() rather than `.highest("value")`?

Reply:
Some addon developper don't use "value" but "add" or "sub" if i use symmetric with highest("value") then they will must modified the heals ability for what this function.

Reply:
In that case, isn't it wrong to use `"value"` for healer_down? The argument that ability definitions may use `add` or `sub` also applies to negative healing.

Reply:
i will work but use "value" is not wrong, but incomplete, i must add lowest("add") and lowest("sub") or highest("sub").

Reply:
I appreciate your point, but don't reimplement the logic for computing the "net value" of an ability. The computation of healer_down and healer_up should be symmetric. In the PR's current state the computation _is_ symmetric, but it doesn't call get_composite_value at all now, which is going to break any heals ability definition that doesn't use `"value"`.